vtex-apps/b2b-orders-history
# Fix/order actions

#### What problem is this solving?

This PR depends on https://github.com/vtex-apps/b2b-organizations-graphql/pull/38

In the current version of this app, many links point to the normal order history pages. In other words it's easy for the user to accidentally leave the B2B order history context, which for example would result in them not being able to view an order placed by someone else within their organization. 

This PR updates all of the UI links to ensure that the user stays within the `/orders-history/` path (and not `/orders/`). 

Additionally, I have determined that the "Order Again" and "Change Order" APIs can only be called by the user who originally placed the order (or a call center operator impersonating them). Therefore, this PR adds logic to hide these buttons if the current user is viewing an order who was placed by someone else within their organization. 

Finally, this PR adds logic to call the B2B version of the `user-cancel-request` endpoint, which is now provided by `b2b-organizations-graphql`. This endpoint allows cancellation requests to be submitted by users other than the user who originally placed the order, as long as they are part of the same organization.  

#### How should this be manually tested?

This new version and a new version of `b2b-organizations-graphql` are both linked in http://b2bsuite--sandboxusdev.myvtex.com/

Log in as an organization user, go to My Account -> Orders (there are two Orders links, choose the second one) and try viewing and cancelling an order. Also try changing the URL to an order ID that you should not be able to access.
